pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -178,17 +180,56 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 	return utilerrors.NewAggregate(errs)
 }
 
+// validateCertificateSAN makes sure that a given certificate is valid
+// at least for the SANs defined in the configuration.
+func (c *EtcdCertSignerController) validateCertificateSAN(certName string, ipAddresses[] string) error {
+	secret, err := c.secretLister.Secrets(operatorclient.TargetNamespace).Get(certName)
+	if err != nil {
+		return fmt.Errorf("Could not get secrets from %s: %w",certName, err)
+	}
+	if secret.Data == nil {
+		return fmt.Errorf("secret data not found in %s", certName)
+	}
+	tlsCrt, ok := secret.Data["tls.crt"]
+	if !ok {
+		return fmt.Errorf("TLS cert data not found in %s", certName)
+	}
+	if len(tlsCrt) == 0 {
+		return fmt.Errorf("%s MISSING tls.crt content\n", certName)
+	}
+	certificates, err := cert.ParseCertsPEM([]byte(tlsCrt))
+	if err != nil {
+		return fmt.Errorf("Could not parse TLS cert for %s: %w", certName, err)
```

Comment:
nit: we generally follow the go convention[1], error messages that should not be capitalized.

[1]https://github.com/golang/go/wiki/CodeReviewComments#error-strings